core/src/integrationTest/java/tech/pegasys/eth2signer/core/DirectoryBackedArtifactSigningProviderIntegrationTest.java

```diff
@@ -0,0 +1,160 @@
+    signingMetadata.put("keyPath", "/v1/secret/data/secretPath");
+    signingMetadata.put("token", "accessToken");
+    signingMetadata.put("tlsEnabled", "true");
+    signingMetadata.put("tlsKnownServersPath", malformedknownServers.toString());
+    YAML_OBJECT_MAPPER.writeValue(filename.toFile(), signingMetadata);
+
+    final Optional<ArtifactSigner> signer = signerProvider.getSigner(PUBLIC_KEY);
+    assertThat(signer).isEmpty();
+
+    assertThat(logAppender.getLogMessagesReceived().size()).isNotZero();
+    final List<String> errorMsgs =
+        logAppender.getLogMessagesReceived().stream()
+            .filter(logEvent -> logEvent.getLevel().equals(Level.ERROR))
+            .map(logEvent -> logEvent.getMessage().getFormattedMessage())
+            .collect(Collectors.toList());
+    assertThat(errorMsgs.size()).isEqualTo(2);
+    assertThat(errorMsgs.get(0))
+        .contains("Error parsing signing metadata file " + filename.getFileName());
+    assertThat(errorMsgs.get(0)).contains("Invalid fingerprint");
+    assertThat(errorMsgs.get(1))
+        .contains("No valid matching metadata file found for the identifier " + PUBLIC_KEY);
+  }
+
+  @Test
+  void missingHashicorpFingerprintFileWhichCannotBeCreatedShowsUsefulErrorLog() throws IOException {
+    final Path filename = configsDirectory.resolve(PUBLIC_KEY + "." + FILE_EXTENSION);
+
+    final Path missingFingerprintFile = configsDirectory.resolve("missingFingerprintFile");
+
+    try {
+      final Map<String, String> signingMetadata = new HashMap<>();
+      signingMetadata.put("type", "hashicorp");
+      signingMetadata.put("serverHost", "localhost");
+      signingMetadata.put("keyPath", "/v1/secret/data/secretPath");
+      signingMetadata.put("token", "accessToken");
+      signingMetadata.put("tlsEnabled", "true");
+      signingMetadata.put("tlsKnownServersPath", missingFingerprintFile.toString());
+      YAML_OBJECT_MAPPER.writeValue(filename.toFile(), signingMetadata);
+
+      configsDirectory.toFile().setWritable(false);
+      final Optional<ArtifactSigner> signer = signerProvider.getSigner(PUBLIC_KEY);
+      assertThat(signer).isEmpty();
+
+    } finally {
+      configsDirectory.toFile().setWritable(true);
+    }
+  }
+
+  @Test
+  void hashicorpVaultNotRunningProducesErrorMessageIndicatingFailure() throws IOException {
+    final Path filename = configsDirectory.resolve(PUBLIC_KEY + "." + FILE_EXTENSION);
+    final Map<String, String> signingMetadata = new HashMap<>();
+    signingMetadata.put("type", "hashicorp");
+    signingMetadata.put("serverHost", "localhost");
+    signingMetadata.put("keyPath", "/v1/secret/data/secretPath");
+    signingMetadata.put("token", "accessToken");
+    signingMetadata.put("tlsEnabled", "false");
+    YAML_OBJECT_MAPPER.writeValue(filename.toFile(), signingMetadata);
+
+    final Optional<ArtifactSigner> signer = signerProvider.getSigner(PUBLIC_KEY);
+    assertThat(signer).isEmpty();
```

Comment:
Similar to above, this test implies we are checking the error message but there is not assertion for the error message

Reply:
done